seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceReader.java

```diff
@@ -150,8 +160,27 @@ public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
                                                             recordList) {
 
                                                         try {
-                                                            deserializationSchema.deserialize(
-                                                                    record.value(), output);
+                                                            if (StringUtils.isBlank(rowDelimiter)) {
```

Comment:
You are right, the premise is that the content of the message cannot contain newline symbols, otherwise there will be wrong parsing, in fact, the same problem will occur in the text format.

Reply:
I also believe it's best to add this feature to TextDeserializationSchema.